src/tests.rs

```diff
@@ -1,99 +1,130 @@
 #[cfg(test)]
 mod unit {
-    fn check(data: &[u8]) -> Result<Vec<crate::Item>, std::io::Error> {
+    use crate::Error;
+
+    fn check_slice(mut data: &[u8]) -> Result<Vec<crate::Item>, Error> {
```

Comment:
I think these helpers should be after the `#[test]`'s that rely on them, and `check_slice` after `check_both` to match [top down ordering](https://github.com/rustls/rustls/blob/main/CONTRIBUTING.md#top-down-ordering-within-modules)